tests/test_scalars.py

```diff
@@ -339,147 +381,219 @@ def test_timedelta() -> None:
     check(assert_type(td == np.timedelta64(1, "D"), bool), bool)
+    check(assert_type(td == ndarray_td64, np_ndarray_bool), np.ndarray)
     check(
-        assert_type(td == np.array([1, 2, 3], dtype="timedelta64[D]"), np.ndarray),
+        assert_type(td == pd.TimedeltaIndex([1, 2, 3], unit="D"), np_ndarray_bool),
         np.ndarray,
     )
+    check(assert_type(td == pd.Series([1, 2, 3]), pd.Series), pd.Series)
+    check(assert_type(dt.timedelta(days=1) == td, bool), bool)
     check(
-        assert_type(td == pd.TimedeltaIndex([1, 2, 3], unit="D"), np.ndarray),
+        assert_type(pd.TimedeltaIndex([1, 2, 3], unit="D") == td, np_ndarray_bool),
         np.ndarray,
     )
-    check(assert_type(td == pd.Series([1, 2, 3]), pd.Series), pd.Series)
-    check(assert_type(td == pd.Timestamp("2016-01-01"), bool), bool)
+    check(assert_type(pd.Series([1, 2, 3]) == td, "pd.Series[bool]"), pd.Series)
+
     check(assert_type(td == 1, bool), bool)
     check(assert_type(td == (3 + 2j), bool), bool)
 
     check(assert_type(td != td, bool), bool)
     check(assert_type(td != dt.timedelta(days=1), bool), bool)
     check(assert_type(td != np.timedelta64(1, "D"), bool), bool)
+    check(assert_type(td != ndarray_td64, np_ndarray_bool), np.ndarray)
     check(
-        assert_type(td != np.array([1, 2, 3], dtype="timedelta64[D]"), np.ndarray),
+        assert_type(td != pd.TimedeltaIndex([1, 2, 3], unit="D"), np_ndarray_bool),
         np.ndarray,
     )
+    check(assert_type(td != pd.Series([1, 2, 3]), pd.Series), pd.Series)
+    check(assert_type(dt.timedelta(days=1) != td, bool), bool)
     check(
-        assert_type(td != pd.TimedeltaIndex([1, 2, 3], unit="D"), np.ndarray),
+        assert_type(pd.TimedeltaIndex([1, 2, 3], unit="D") != td, np_ndarray_bool),
         np.ndarray,
     )
-    check(assert_type(td != pd.Series([1, 2, 3]), pd.Series), pd.Series)
-    check(assert_type(td != pd.Timestamp("2016-01-01"), bool), bool)
+    check(assert_type(pd.Series([1, 2, 3]) != td, "pd.Series[bool]"), pd.Series)
     check(assert_type(td != 1, bool), bool)
     check(assert_type(td != (3 + 2j), bool), bool)
 
+    # Mismatch due to NumPy ops returning Any
+    check(assert_type(np.array([1, 2, 3]) * td, Any), np.ndarray)
+    check(assert_type(np.array([1.2, 2.2, 3.4]) * td, Any), np.ndarray)
+    check(assert_type(np.timedelta64(1, "D") < td, Any), np.bool_)
+    check(assert_type(np.timedelta64(1, "D") > td, Any), np.bool_)
+    check(assert_type(np.timedelta64(1, "D") <= td, Any), np.bool_)
+    check(assert_type(np.timedelta64(1, "D") >= td, Any), np.bool_)
+    check(assert_type(np.timedelta64(1, "D") == td, Any), np.bool_)
+    check(assert_type(ndarray_td64 == td, Any), np.ndarray)
+    check(assert_type(ndarray_td64 != td, Any), np.ndarray)
+    check(assert_type(np.timedelta64(1, "D") != td, Any), np.bool_)
+
 
 def test_interval() -> None:
     i0 = pd.Interval(0, 1, closed="left")
```

Comment:
these tests would be easier to follow by naming `i0`, `i1`, `i2` and `i3` to be `i_int`, `i_float`, `i_ts` and `i_td` respectively